src/widgets/shared-utils.tsx

```diff
@@ -1,6 +1,10 @@
 import { newWorkspaceItem } from "@openmrs/esm-api";
 
-export const openWorkspaceTab = (componentToAdd, componentName, params?) => {
+export function openWorkspaceTab<TProps, TParams = any>(
```

Comment:
Maybe use `TProps = DataCaptureComponentProps` (i.e., use `DataCaptureComponentProps` as default).

Below you state that the functional component has props that are either derived into `TProps` or are `DataCaptureComponentProps`; which means it will always just be `TProps` and in some cases `TProps = DataCaptureComponentProps`. The above takes that away with a default value.

Reply:
That's very clear, thanks. I've changed it to be so.